src/fu-engine.c

```diff
@@ -742,67 +742,87 @@ fu_engine_reset_config(FuEngine *self, const gchar *section, GError **error)
+fu_engine_modify_config(FuEngine *self,
+			const gchar *section,
+			const gchar *key,
+			const gchar *value,
+			GError **error)
 {
 	g_auto(GStrv) section_key = NULL;
-	const gchar *keys[] = {"ArchiveSizeMax",
-			       "AllowEmulation",
-			       "ApprovedFirmware",
-			       "BlockedFirmware",
-			       "DisabledDevices",
-			       "DisabledPlugins",
-			       "EnumerateAllDevices",
-			       "EspLocation",
-			       "HostBkc",
-			       "IdleTimeout",
-			       "IgnorePower",
-			       "OnlyTrusted",
-			       "P2pPolicy",
-			       "ReleaseDedupe",
-			       "ReleasePriority",
-			       "ShowDevicePrivate",
-			       "TestDevices",
-			       "TrustedReports",
-			       "TrustedUids",
-			       "UpdateMotd",
-			       "UriSchemes",
-			       "VerboseDomains",
-			       "test:AnotherWriteRequired",
-			       "test:CompositeChild",
-			       "test:DecompressDelay",
-			       "test:NeedsActivation",
-			       "test:NeedsReboot",
-			       "test:RegistrationSupported",
-			       "test:RequestDelay",
-			       "test:RequestSupported",
-			       "test:VerifyDelay",
-			       "test:WriteDelay",
-			       "test:WriteSupported",
-			       NULL};
 
 	g_return_val_if_fail(FU_IS_ENGINE(self), FALSE);
+	g_return_val_if_fail(section != NULL, FALSE);
 	g_return_val_if_fail(key != NULL, FALSE);
 	g_return_val_if_fail(value != NULL, FALSE);
 	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
 
 	/* check keys are valid */
-	if (!g_strv_contains(keys, key)) {
-		g_set_error(error, FWUPD_ERROR, FWUPD_ERROR_NOT_FOUND, "key %s not supported", key);
+	if (g_strcmp0(section, "fwupd") == 0) {
+		const gchar *keys[] = {"ArchiveSizeMax",
+				       "AllowEmulation",
+				       "ApprovedFirmware",
+				       "BlockedFirmware",
+				       "DisabledDevices",
+				       "DisabledPlugins",
+				       "EnumerateAllDevices",
+				       "EspLocation",
+				       "HostBkc",
+				       "IdleTimeout",
+				       "IgnorePower",
+				       "OnlyTrusted",
+				       "P2pPolicy",
+				       "ReleaseDedupe",
+				       "ReleasePriority",
+				       "ShowDevicePrivate",
+				       "TestDevices",
+				       "TrustedReports",
+				       "TrustedUids",
+				       "UpdateMotd",
+				       "UriSchemes",
+				       "VerboseDomains",
+				       NULL};
+		if (!g_strv_contains(keys, key)) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_FOUND,
+				    "key %s not supported for [%s]",
+				    key,
+				    section);
+			return FALSE;
+		}
+	} else if (g_strcmp0(section, "test") == 0) {
+		const gchar *keys[] = {"AnotherWriteRequired",
+				       "CompositeChild",
+				       "DecompressDelay",
+				       "NeedsActivation",
+				       "NeedsReboot",
+				       "RegistrationSupported",
+				       "RequestDelay",
+				       "RequestSupported",
+				       "VerifyDelay",
+				       "WriteDelay",
+				       "WriteSupported",
+				       NULL};
```

Comment:
to me this is begging for a vfunc to call into a plugin rather than storing in the daemon now.